src/patient-search/patient-search.component.tsx

```diff
@@ -1,10 +1,165 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { match } from "react-router";
+import { debounce, isEmpty } from "lodash";
+import { performPatientSearch } from "./patient-search.resource";
+import styles from "./patient-search.component.css";
+import PatientSearchResults from "../patient-search-result/patient-search-result.component";
+import { object } from "prop-types";
 
 export default function PatientSearch(props: PatientSearchProps) {
-  return <section>The patient search!</section>;
+  const searchTimeout = 300;
+  const resultsPerPage = 10;
+
+  const [searchResults, setSearchResults] = useState([]);
+  const [emptyResult, setEmptyResult] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const [pagedResults, setPagedResults] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [showNextButton, setShowNextButton] = useState(false);
+  const [showPreviousButton, setShowPreviousButton] = useState(false);
+
+  useEffect(() => {
+    currentPage * resultsPerPage > searchResults.length
+      ? setShowNextButton(false)
+      : setShowNextButton(true);
+    currentPage !== 1
+      ? setShowPreviousButton(true)
+      : setShowPreviousButton(false);
+  }, [pagedResults, currentPage]);
```

Comment:
For completeness, you should add `resultsPerPage` to the conditions array here. There isn't a bug here, but your code violates the strict rules in place for how to create the conditions array ([details](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects)).

The only situation it would end up mattering is if we make `resultsPerPage` changeable. Which is unlikely. But it's good practice to just follow the rules that are in place about the conditions array, to get used to it.